### ISSUE TITLE
VJP non-differentiable-input fallback: code returns NaN but comment says "zero"

### ISSUE THREAD

**Issue by andrinr** [pasteurlabs/tesseract-jax#187]
### Description

In `tesseract_jax/tesseract_compat.py:215-234`, the VJP fallback for non-differentiable non-static inputs returns `np.nan` but the comment one line above says "return zero gradient":

```python
elif (
    tan_idx < len(has_tangent)
    and not is_static_mask[all_idx]
    and not has_tangent[tan_idx]
):
    # Non-differentiable but non-static input: return zero gradient
    # with the same shape/dtype as the corresponding input array
    out.append(
        np.full(
            array_args[array_idx].shape,
            np.nan,
            dtype=array_args[array_idx].dtype,
        )
    )
```

The JVP path at line 161 also returns NaN for non-differentiable outputs, and that behaviour is tested at `tests/test_api.py:225` (`assert jnp.all(jnp.isnan(jvp["metadata"]))`), so NaN may be the intended sentinel. If so, the comment should be updated to say so. If zero was intended, the code should use zero — zero composes cleanly downstream, NaN poisons everything that touches it.

### Steps to reproduce

A schema declares `x` and `y` both as `Differentiable`. The caller requests a gradient only for `x`, passing `y` as a primal. `tesseract_jax`'s VJP dispatch then takes this branch for `y`, and the entry in the gradient tuple for `y` is filled with NaN of the same shape as `y`.

### Logs

n/a — observed by reading the source.

### OS

Linux

### Tesseract + Tesseract-JAX version

tesseract-core 1.8.0, tesseract-jax 0.3.0

**Comment by andrinr** [comment]
For posterity, after closing — what I verified before this was closed:

The comment / code mismatch at `tesseract_compat.py:225` (comment says "zero gradient") versus `:228-232` (code fills `np.nan`) is real on `main`. But I couldn't surface that NaN to user-visible state via any path I tried:

- `jax.grad(f)(x)` with `y` closed over — `y` has no slot in the user's gradient, so the dispatcher's NaN at the `y` position never reaches the user.
- `jax.vjp(f, {x, y})` — both have tangents requested → the elif branch isn't taken, no NaN at all.
- `jax.vjp(f, {x, stop_gradient(y)})` — JAX produces an `ad.Zero` cotangent for the stopped input; user-visible `y` gradient comes back as `0.0`, not NaN.

The branch fills a positional slot that JAX's transpose rule builds for tuple-length contract reasons but doesn't actually read for any user-requested derivative. So this is a cosmetic / dead-placeholder issue, not a user-visible bug. Filed the narrower drive-by doc fix separately.
